include/cinatra/ylt/coro_io/coro_file.hpp

```diff
@@ -158,24 +162,82 @@ class coro_file {
     return size;
   }
 
-#if defined(__GNUC__)
   bool open_fd(std::string_view filepath, int open_mode = flags::read_write) {
     if (fd_file_) {
       return true;
     }
 
+#if defined(ASIO_WINDOWS)
+    int fd = _open(filepath.data(), adjust_open_mode(open_mode));
+#else
     int fd = open(filepath.data(), open_mode);
+#endif
     if (fd < 0) {
       return false;
     }
 
     fd_file_ = std::shared_ptr<int>(new int(fd), [](int* ptr) {
+#if defined(ASIO_WINDOWS)
+      _close(*ptr);
+#else
       ::close(*ptr);
+#endif
       delete ptr;
     });
     return true;
   }
 
+#if defined(ASIO_WINDOWS)
+  static int adjust_open_mode(int open_mode) {
+    switch (open_mode) {
+      case flags::read_only:
+        return _O_RDONLY;
+      case flags::write_only:
+        return _O_WRONLY;
+      case flags::read_write:
+        return _O_RDWR;
+      case flags::append:
+        return _O_APPEND;
+      case flags::create:
+        return _O_CREAT;
+      case flags::exclusive:
+        return _O_EXCL;
+      case flags::truncate:
+        return _O_TRUNC;
+      case flags::create_write:
+        return _O_CREAT | _O_WRONLY;
+      case flags::create_write_trunc:
+        return _O_CREAT | _O_WRONLY | _O_TRUNC;
+      case flags::create_read_write_trunc:
+        return _O_RDWR | _O_CREAT | _O_TRUNC;
+      case flags::create_read_write_append:
+        return _O_RDWR | _O_CREAT | _O_APPEND;
+      case flags::sync_all_on_write:
+      default:
+        return open_mode;
+        break;
+    }
+    return open_mode;
+  }
+
+  static ssize_t pread(int fd, void* buf, size_t count, uint64_t offset) {
+    DWORD bytes_read = 0;
+    OVERLAPPED overlapped;
+    memset(&overlapped, 0, sizeof(OVERLAPPED));
+    overlapped.Offset = offset & 0xFFFFFFFF;
+    overlapped.OffsetHigh = (offset >> 32) & 0xFFFFFFFF;
+
+    BOOL ok = ReadFile(reinterpret_cast<HANDLE>(_get_osfhandle(fd)), buf, count,
+                       &bytes_read, &overlapped);
+
+    if (!ok && (errno = GetLastError()) != ERROR_HANDLE_EOF) {
+      return -1;
+    }
+
+    return bytes_read;
+  }
```

Comment:
add a lazy async_pread for windows, you can put this code into async_pread with coro_io::post.

Reply:
after that we can remove __GNUC__